src/main/java/org/jabref/gui/entryeditor/RelatedArticlesTab.java

```diff
@@ -86,6 +165,11 @@ public boolean shouldShow(BibEntry entry) {
 
     @Override
     protected void bindToEntry(BibEntry entry) {
-        setContent(getPane(entry));
+        // Ask for consent to send data to Mr. DLib on first time to tab
+        if (JabRefPreferences.getInstance().getBoolean(JabRefPreferences.ACCEPT_RECOMMENDATIONS)) {
```

Comment:
Hi Christoph, thanks for your help.

I had been using EntryEditorPreferences, but switched because the object doesn't get updated until the application is restarted. This resulted in the dialog still appearing after accepting, until the application was restarted.

Is there a way of doing this such that the preferences are refreshed on each load of the tab?